src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs

```diff
@@ -171,123 +177,249 @@ public enum PopupPositionerConstraintAdjustment
+        /// FlipY adjustment will be the one before the adjustment.
+        /// </remarks>
         FlipY = 8,
+
         All = SlideX|SlideY|FlipX|FlipY
     }
 
     static class PopupPositioningEdgeHelper
     {
-        public static void ValidateEdge(this PopupPositioningEdge edge)
+        public static void ValidateEdge(this PopupAnchor edge)
         {
-            if (((edge & PopupPositioningEdge.Left) != 0 && (edge & PopupPositioningEdge.Right) != 0)
+            if (((edge & PopupAnchor.Left) != 0 && (edge & PopupAnchor.Right) != 0)
                 ||
-                ((edge & PopupPositioningEdge.Top) != 0 && (edge & PopupPositioningEdge.Bottom) != 0))
+                ((edge & PopupAnchor.Top) != 0 && (edge & PopupAnchor.Bottom) != 0))
                 throw new ArgumentException("Opposite edges specified");
         }
 
-        public static PopupPositioningEdge Flip(this PopupPositioningEdge edge)
+        public static void ValidateGravity(this PopupGravity gravity)
+        {
+            ValidateEdge((PopupAnchor)gravity);
+        }
+
+        public static PopupAnchor Flip(this PopupAnchor edge)
         {
-            var hmask = PopupPositioningEdge.Left | PopupPositioningEdge.Right;
-            var vmask = PopupPositioningEdge.Top | PopupPositioningEdge.Bottom;
+            var hmask = PopupAnchor.Left | PopupAnchor.Right;
+            var vmask = PopupAnchor.Top | PopupAnchor.Bottom;
             if ((edge & hmask) != 0)
                 edge ^= hmask;
             if ((edge & vmask) != 0)
                 edge ^= vmask;
             return edge;
         }
 
-        public static PopupPositioningEdge FlipX(this PopupPositioningEdge edge)
+        public static PopupAnchor FlipX(this PopupAnchor edge)
         {
-            if ((edge & PopupPositioningEdge.HorizontalMask) != 0)
-                edge ^= PopupPositioningEdge.HorizontalMask;
+            if ((edge & PopupAnchor.HorizontalMask) != 0)
+                edge ^= PopupAnchor.HorizontalMask;
             return edge;
         }
         
-        public static PopupPositioningEdge FlipY(this PopupPositioningEdge edge)
+        public static PopupAnchor FlipY(this PopupAnchor edge)
         {
-            if ((edge & PopupPositioningEdge.VerticalMask) != 0)
-                edge ^= PopupPositioningEdge.VerticalMask;
+            if ((edge & PopupAnchor.VerticalMask) != 0)
+                edge ^= PopupAnchor.VerticalMask;
             return edge;
         }
-        
+
+        public static PopupGravity FlipX(this PopupGravity gravity)
+        {
+            return (PopupGravity)FlipX((PopupAnchor)gravity);
+        }
+
+        public static PopupGravity FlipY(this PopupGravity gravity)
+        {
+            return (PopupGravity)FlipY((PopupAnchor)gravity);
+        }
     }
 
+    /// <summary>
+    /// Defines the edges around an anchor rectangle on which a popup will open.
+    /// </summary>
     [Flags]
-    public enum PopupPositioningEdge
```

Comment:
@kekekeks I split this enum into two separate enums, because it allows us to more easily explain what the enum values do in the case of anchors and gravity than when it's a single enum. I notice that in the xdg spec it's two enums anyway.

Reply:
I know this is a breaking change, but IMO we should be doing breaking changes now if it makes things easier to understand. I had difficulty understanding what "gravity" was, I think this makes it clearer.